petl/io/gsheet.py

```diff
@@ -0,0 +1,158 @@
+    module: http://gspread.readthedocs.io/en/latest/index.html
+
+    """
+
+    return GoogleSheetView(filename,
+                           credentials,
+                           forcename=forcename,
+                           worksheet_title=worksheet_title,
+                           range_string=range_string)
+
+
+class GoogleSheetView(Table):
+    """This module resembles XLSXView."""
+
+    def __init__(self, filename, credentials, forcename, worksheet_title,
+                 range_string):
+        self.filename = filename
+        self.credentials = credentials
+        self.forcename = forcename
+        self.worksheet_title = worksheet_title
+        self.range_string = range_string
+
+    def __iter__(self):
+        import gspread
+        gspread_client = gspread.authorize(self.credentials)
+        if self.forcename:
+            wb = gspread_client.open(self.filename)
+        else:
+            try:
+                wb = gspread_client.open_by_key(self.filename)
+            except gspread.exceptions.SpreadsheetNotFound:
+                wb = gspread_client.open(self.filename)
+
+        # Allow for user to specify no sheet, sheet index or sheet name
+        if self.worksheet_title is None:
+            ws = wb.sheet1
+        elif isinstance(self.worksheet_title, int):
+            ws = wb.get_worksheet(self.worksheet_title)
+        else:
+            # use text_type for cross version compatibility
+            ws = wb.worksheet(text_type(self.worksheet_title))
+
+        # grab the range or grab the whole sheet
+        if self.range_string:
+            # start_cell -> top left, end_cell -> bottom right
+            start_cell, end_cell = self.range_string.split(':')
+            start_row, start_col = gspread.utils.a1_to_rowcol(start_cell)
+            end_row, end_col = gspread.utils.a1_to_rowcol(end_cell)
+            # gspread starts its indices at 1
+            for i, row in enumerate(ws.get_all_values(), start=1):
+                if i in range(start_row, end_row + 1):
+                    start_col_index = start_col - 1
+                    yield tuple(row[start_col_index:end_col])
+        else:
+            # no range specified, so return all the rows
+            for row in ws.get_all_values():
+                yield tuple(row)
+
+
+def togsheet(tbl, filename, credentials, worksheet_title=None,
+             share_emails=[], role='writer'):
+    """
+    Write a table to a new google sheet.
+
+    filename will be the title of the sheet when uploaded to google sheets.
+
+    credentials are used to authenticate with the google apis.
+    For more info, visit: http://gspread.readthedocs.io/en/latest/oauth2.html
+
+    If worksheet_title is specified, the first worksheet in the spreadsheet
+    will be renamed to the value of worksheet_title.
+
+    The spreadsheet will be shared with all emails in `share_emails` with
+    `role` permissions granted.
+
+    set the permissions of all `share_emails` as `role`.
```

Comment:
This line looks redundant, `role` is already described in the preceding sentence.